### PR TITLE
Blog page tweaks

### PR DESCRIPTION
**Pull request by arshdkhn1**
blog-with-toggle

### CODE REVIEW

File: static/css/janitor-new.css [hunk 1]
```diff
@@ -776,23 +798,46 @@ textarea.form-control {
 
 /* Blog page */
 
-.blog {
-  font-size: 18px;
-  line-height: 1.6;
+.blog  h1.heading {
```
Comment (notriddle) [inline static/css/janitor-new.css]: two spaces?

Comment (arshdkhn1) [reply static/css/janitor-new.css]: fixed

File: static/css/janitor-new.css [hunk 2]
```diff
@@ -776,23 +798,90 @@ textarea.form-control {
...
-.blog article:first-child {
-  border: none;
-  padding-top: 0;
+.blog .article-wrapper::after {
+  content: ' ';
+  position: absolute;
+  width: 100%;
+  bottom: 0;
+  /* fallback */
+  background: #fff;
```
Comment (nt1m) [inline static/css/janitor-new.css]: No need for the fallback, linear-gradients are fairly well supported.

File: static/css/janitor-new.css [hunk 3]
```diff
@@ -776,23 +798,90 @@ textarea.form-control {
...
+  overflow: auto;
+  height: 400px;
+  overflow: hidden;
 }
 
-.blog article:first-child {
-  border: none;
-  padding-top: 0;
+.blog .article-wrapper::after {
+  content: ' ';
```
Comment (nt1m) [inline static/css/janitor-new.css]: nit: `content: "";` (double quotes, and no space inside)

File: static/css/janitor-new.css [hunk 4]
```diff
@@ -776,23 +798,90 @@ textarea.form-control {
...
+}
+
+.blog blockquote {
+  border-left: solid 4px;
+  background: #f9f9f9;
+  padding: 20px;
+  margin: 20px 0;
+}
+
+.blog blockquote > :last-child {
```
Comment (nt1m) [inline static/css/janitor-new.css]: Is this syntax legal ? I thought you needed to do `.blog blockquote > *:last-child`

File: static/css/janitor-new.css [hunk 5]
```diff
@@ -776,23 +798,90 @@ textarea.form-control {
...
+}
+
+.blog article h2 {
+  font-weight: 700;
+  font-size: 18px;
+}
+
+.blog article p img {
+  width: 100%;
+  height: 100%;
```
Comment (nt1m) [inline static/css/janitor-new.css]: Why height: 100%; ? Won't that cause the images to be stretched ?

File: static/css/janitor-new.css [hunk 6]
```diff
@@ -776,23 +798,90 @@ textarea.form-control {
...
+  margin-bottom: 60px;
+}
+
+.blog article h1 {
+  margin-bottom: 40px;
+}
+
+.blog article p {
+  color: #0b0b0d;
+  line-height: 24px;
```
Comment (nt1m) [inline static/css/janitor-new.css]: hmm, isn't `line-height: 1.5` on `p` enough ?

File: static/css/janitor-new.css [hunk 7]
```diff
@@ -30,6 +30,10 @@ body {
   float: right;
 }
 
+.text-center {
```
Comment (nt1m) [inline static/css/janitor-new.css]: Why did you move this up ? This, `.bold` and `.heading` should be in the typography section.

File: static/css/janitor-new.css [hunk 8]
```diff
@@ -63,6 +67,11 @@ body {
   padding: 1em;
 }
 
+.heading {
+  font-weight: 200;
```
Comment (nt1m) [inline static/css/janitor-new.css]: This should be under the typography section of the CSS.

**Comment by nt1m** [comment]
Like @notriddle, I'm not sure I like the hugely spaced centered "blog" heading. It's just not consistent with anything else on the website, and also doesn't fit well with the header.

File: static/css/janitor-new.css (hunk 3 above)
Comment (arshdkhn1) [reply static/css/janitor-new.css]: Is this a css rule?

File: static/css/janitor-new.css (hunk 7 above)
Comment (arshdkhn1) [reply static/css/janitor-new.css]: i thought text-cente is more of a helper class.

File: static/css/janitor-new.css (hunk 8 above)
Comment (arshdkhn1) [reply static/css/janitor-new.css]: done

File: static/css/janitor-new.css (hunk 6 above)
Comment (arshdkhn1) [reply static/css/janitor-new.css]: I was thinking of giving lh:2 but used 24px. Imo blog have lot p tags so big lineheight makes them more readable.

File: static/css/janitor-new.css (hunk 4 above)
Comment (arshdkhn1) [reply static/css/janitor-new.css]: fixed

File: static/css/janitor-new.css (hunk 2 above)
Comment (arshdkhn1) [reply static/css/janitor-new.css]: can i just let the background #fff be there for opera users :P

File: static/css/janitor-new.css (hunk 5 above)
Comment (arshdkhn1) [reply static/css/janitor-new.css]: image has its original aspect ratio but I must say it will not look good for all images.

File: static/css/janitor-new.css (hunk 2 above)
Comment (nt1m) [reply static/css/janitor-new.css]: The deprecated opera 12 version ? naah

File: static/css/janitor-new.css (hunk 3 above)
Comment (nt1m) [reply static/css/janitor-new.css]: We use double quotes throughout the CSS, let's be consistent here.

File: static/css/janitor-new.css (hunk 5 above)
Comment (nt1m) [reply static/css/janitor-new.css]: Is width: 100%; enough ?

File: static/css/janitor-new.css (hunk 2 above)
Comment (arshdkhn1) [reply static/css/janitor-new.css]: ohk :(

File: static/css/janitor-new.css (hunk 5 above)
Comment (arshdkhn1) [reply static/css/janitor-new.css]: image looks to have original aspect ratio. I dont undestand what u are asking? Anyway, I have changed these css.

**Comment by arshdkhn1** [comment]
@nt1m Other pages that I made doesn't need .heading. Landing page and some other page will. I think it looks good.

**Comment by jankeromnes** [comment]
Thanks a lot for improving the new blog page!

Please note that this page needs a few more small improvements, listed in #219.

I'm not sure about the collapsed blog posts (I don't like that the content you came for is hidden by default and requires an extra click). I'm also not sure about the hugely spaced & centered "Blog" title (I don't think it's really needed, because it's already pretty obvious what this page is about). But I'll defer this review to @notriddle and @nt1m, who are much better reviewers than me for this.

But I'm happy to help in any other way 😄 (e.g. I'm happy to answer any question, or give my unskilled opinion on alternative design suggestions, etc).

**Comment by nt1m** [comment]
@arshdkhn1 I would honestly just remove the huge "Blog" heading, as @jankeromnes says, it's quite obvious what the page is. What do you think ?

I like the typography improvements and the toggleable posts however.

**Comment by arshdkhn1** [comment]
@jankeromnes blogs are one above each other and for user to goto some specific blog he has to scroll too much..

**Comment by jankeromnes** [comment]
> @jankeromnes blogs are one above each other and for user to goto some specific blog he has to scroll too much..

As I mentioned, I really don't like having to make an additional click to get the content I came for (e.g. if I see a link like cool.website/#new-blog-post I think "well I can have a look" but if the page gives me 3 lines and a button to see more, I'm going to be too lazy to click and close the tab without reading).

And I can quickly find any blog post using Cmd+F, without spending any time scrolling. And I feel like a side menu would work better to help users who don't use Cmd+F find any post.

But @nt1m seems to like the collapsed blog posts, and I deferred my review to him, so he gets to decide. 😛

Still, maybe we can always have at least one blog post expanded by default? E.g. the latest blog post is expanded but all others are collapsed so you don't have to scroll too much. Or, you clicked on an anchor like `#janitor-news-4` so we know you want to read that post so we pre-expand it for you.

**Comment by arshdkhn1** [comment]
@jankeromnes makes sense.

File: templates/blog-new.html [hunk 9]
```diff
@@ -2,11 +2,17 @@
...
-      <footer><a class="btn" href="{{= comments_url in xmlattr}}">{{= comments_count in html}} comments</a></footer>{{
+      <div class="blog-post">
+        <input type="checkbox" class="blog-toggle" id="{{= slug in xmlattr}}-cb">
+        <label for="{{= slug in xmlattr}}-cb" title="Toggle"></label>
+        <div class="article-wrapper">
+          <article>
+            <h1 id="{{= slug in xmlattr}}">{{= title in html}}</h1>
+            {{= post_body_html}}
+          </article>
+          <footer><a class="btn" href="{{= comments_url in xmlattr}}">{{if comments_count then {{{{= comments_count in html}}}} else {{No}}}} comment{{if {{comments_count !== 1}} then {{s}}}}</a></footer>
```
Comment (arshdkhn1) [inline templates/blog-new.html]: why are we doing `{{= variableName in xmlattr}}` instead of `{{= variableName}}` when we have defined all the variables on line 3?

File: static/js/blog-new.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+function maximizeBlog (hash) {
+  if (hash) {
+    document.getElementById(hash.slice(1) + '-cb').checked = true;
+  }
+}
+
+maximizeBlog(window.location.hash);
+
+Array.forEach(document.querySelectorAll('.blog article p a'), element => {
```
Comment (arshdkhn1) [inline static/js/blog-new.js]: Can we use arrow functions(es6) ? @nt1m

Comment (arshdkhn1) [reply static/js/blog-new.js]: I put it here because this belongs to blog page only.

Comment (nt1m) [reply static/js/blog-new.js]: I don't think @jankeromnes wants us to.

Comment (arshdkhn1) [reply static/js/blog-new.js]: janitor-new.js has arrow functions

Comment (nt1m) [reply static/js/blog-new.js]: oh, that must be a mistake then, we should probably put normal functions back

Comment (arshdkhn1) [reply static/js/blog-new.js]: I will change the blog-new.js function. Other function can be converted in another pr(i have added that #257).

File: static/js/blog-new.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
+function maximizeBlog (hash) {
```
Comment (nt1m) [inline static/js/blog-new.js]: expandPost ?

File: static/css/janitor-new.css [hunk 12]
```diff
@@ -776,23 +793,88 @@ textarea.form-control {
...
+.blog blockquote {
+  border-left: solid .25em;
+  background: #fafafa;
+  padding: 20px;
+  margin: 20px 0;
+  font-style: italic;
+}
+
+.blog blockquote > *:last-child {
+  margin-bottom: 0;
```
Comment (nt1m) [inline static/css/janitor-new.css]: I guess this rule isn't useful since you didn't notice it didn't work before.

File: static/js/blog-new.js (hunk 11 above)
Comment (arshdkhn1) [reply static/js/blog-new.js]: ohk i ll rename it

File: static/css/janitor-new.css (hunk 12 above)
Comment (arshdkhn1) [reply static/css/janitor-new.css]: it worked before as well.

File: static/js/blog-new.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+expandPost(window.location.hash);
+
+Array.forEach(document.querySelectorAll('.blog article p a'), function (element) {
+  element.target = '_blank';
+});
+
+Array.forEach(document.querySelectorAll('.blog a.link'), function (element) {
+  element.onclick = expandPost.bind(expandPost, element.getAttribute('href'));
```
Comment (nt1m) [inline static/js/blog-new.js]: This is easier to read IMO:
```js
element.onclick = function () {
  expandPost(element.getAttribute("href"));
};
```

Comment (arshdkhn1) [reply static/js/blog-new.js]: ohk

File: static/js/blog-new.js [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  }
+}
+
+expandPost(window.location.hash);
+
+Array.forEach(document.querySelectorAll('.blog article p a'), function (element) {
+  element.target = '_blank';
+});
+
+Array.forEach(document.querySelectorAll('.blog a.link'), function (element) {
```
Comment (nt1m) [inline static/js/blog-new.js]: Can you change this to `.blog .icon.link` to be more explicit about what this is targeting ?

Comment (arshdkhn1) [reply static/js/blog-new.js]: sure

File: static/js/blog-new.js [hunk 15]
```diff
@@ -0,0 +1,18 @@
+function expandPost (hash) {
+  // If hash is an empty string then expand the first blog
+  var blogId = hash.slice(1) || document.querySelector('.article-wrapper h1').id;
+
```
Comment (nt1m) [inline static/js/blog-new.js]: nit: remove empty line
